Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSubjectHierarchyComboBox.h

```diff
@@ -57,6 +57,26 @@ class Q_SLICER_MODULE_SUBJECTHIERARCHY_WIDGETS_EXPORT qMRMLSubjectHierarchyCombo
   /// Else, the popup tree appears below the combobox, like for a qMRMLNodeComboBox.
   Q_PROPERTY(bool alignPopupVertically READ alignPopupVertically WRITE setAlignPopupVertically)
 
+  /// Filter to show only items that contain any of the given attributes with this name. Empty by default
+  Q_PROPERTY(QStringList includeItemAttributeNamesFilter READ includeItemAttributeNamesFilter WRITE setIncludeItemAttributeNamesFilter)
+  /// Filter to show only items for data nodes that contain any of the given attributes with this name. Empty by default
+  Q_PROPERTY(QStringList includeNodeAttributeNamesFilter READ includeNodeAttributeNamesFilter WRITE setIncludeNodeAttributeNamesFilter)
+  /// Filter to hide items that contain any of the given attributes with this name. Empty by default
+  /// Overrides \sa includeItemAttributeNamesFilter
+  Q_PROPERTY(QStringList excludeItemAttributeNamesFilter READ excludeItemAttributeNamesFilter WRITE setExcludeItemAttributeNamesFilter)
+  /// Filter to hide items for data nodes that contain any of the given attributes with this name. Empty by default
+  /// Overrides \sa includeNodeAttributeNamesFilter
+  Q_PROPERTY(QStringList excludeNodeAttributeNamesFilter READ excludeNodeAttributeNamesFilter WRITE setExcludeNodeAttributeNamesFilter)
+
+  /// Filter to show only items that contain an attribute with this name. Empty by default
+  /// Note: Deprecated, kept only for backwards compatibility. Sets and returns the first attribute in \sa includeNodeAttributeNamesFilter
+  Q_PROPERTY(QString attributeNameFilter READ attributeNameFilter WRITE setAttributeNameFilter)
+  /// Filter to show only items that contain any attribute given in \sa includeItemAttributeNamesFilter with the value.
+  /// If empty, then existence of the attributes is enough to show.
+  /// Exact match is required. Empty by default
+  /// Note: Deprecated, kept only for backwards compatibility. Works consistently with the previous operation.
+  Q_PROPERTY(QString attributeValueFilter READ attributeValueFilter WRITE setAttributeValueFilter)
```

Comment:
We cannot deprecate this. qMRMLNodeCombobox::addAttribute (http://apidocs.slicer.org/master/classqMRMLNodeComboBox.html#a056b54be78c6053b6fd21d51b2b1d045) proven to work be a very simple and powerful way of filtering nodes. In the future we would like to switch over from qMRMLNodeCombobox to qMRMLSubjectHierarchyComboBox (by making qMRMLNodeCombobox an alias to qMRMLSubjectHierarchyComboBox), so it would be important to get closer to feature parity and not get farther from it.

It could be good to have various convenience functions for setting filtering parameters (e.g., set several attribute filters with as a stringlist), but ultimately I think we cannot avoid having filters specified with:
- QString ClassName
- QString AttributeName
- QVariant AttributeValue
- bool Include

The first 3 same as qMRMLNodeComboBox::addAttribute, the last selects between include/exclude as you have nicely implemented.

Reply:
I implemented the filters as you suggest, and added extra functions so that the additional granularity can be used.